lib/ansible/module_utils/cloud.py

```diff
@@ -67,34 +130,28 @@ def found(response_code):
         pass
 
     @classmethod
-    def backoff(cls, tries=10, delay=3, backoff=1.1):
```

Comment:
After thinking on this, I think the best API we can provide is to add a new `jittery_backoff` (or `backoff_with_jitter`) and rename the current `backoff` to `exponential_backoff`, leaving `backoff` as an alias so the current API stays the same. 

I like the factoring internally here though: moving backoff strategies to be generators is really elegant, so preserving those changes would be nice. How about this: first, prefix full_jitter_backoff and exponential_backoff with `_` so it's clear they're not intended for public use. Then make the sig of CloudRetry:

```
class CloudRetry:
    def exponential_backoff(cls, tries=10, delay=3, backoff=1.1)
    def jittered_backoff(cls, retries=10, delay=3, max_delay=60)
    backoff = exponential_backoff
```

Then internally each of those uses the same implementation you've done below, plugging the correct generator in and returning the closure. It shouldn't be too much of a change from this PR, and would preserve the API nicely.